src/pois.rs

```diff
@@ -0,0 +1,243 @@
+            })
+            .unwrap_or_else(|| {
+                let admins = geofinder.get(&self.poi.coord);
+                let country_codes = find_country_codes(iter_admins(&admins));
+                (admins, country_codes)
+            });
+
+        if admins.is_empty() {
+            debug!("The poi {} is not on any admins", &self.poi.id);
+            return None;
+        }
+
+        let zip_codes = match poi_address {
+            Some(mimir::Address::Street(ref s)) => s.zip_codes.clone(),
+            Some(mimir::Address::Addr(ref a)) => a.zip_codes.clone(),
+            _ => vec![],
+        };
+
+        self.poi.administrative_regions = admins;
+        self.poi.address = poi_address;
+        self.poi.label = format_poi_label(
+            &self.poi.name,
+            iter_admins(&self.poi.administrative_regions),
+            &country_codes,
+        );
+        self.poi.labels = format_international_poi_label(
+            &self.poi.names,
+            &self.poi.name,
+            &self.poi.label,
+            iter_admins(&self.poi.administrative_regions),
+            &country_codes,
+            langs,
+        );
+        self.poi.zip_codes = zip_codes;
+        Some(self)
+    }
+}
+
+fn properties_from_row(row: &Row) -> Result<Vec<Property>, String> {
+    let properties = row
+        .try_get::<_, Option<HashMap<_, _>>>("tags")
+        .map_err(|err| {
+            let id: String = row.get("id");
+            warn!("Unable to get tags from row '{}': {:?}", id, err);
+            err.to_string()
+        })?
+        .unwrap_or_else(HashMap::new)
+        .into_iter()
+        .map(|(k, v)| Property {
+            key: k,
+            value: v.unwrap_or_else(|| "".to_string()),
+        })
+        .collect::<Vec<Property>>();
+
+    Ok(properties)
+}
+
+fn build_poi_properties(
+    row: &Row,
+    id: &str,
+    mut properties: Vec<Property>,
+) -> Result<Vec<Property>, String> {
+    let poi_subclass = row.try_get("subclass").map_err(|e| {
+        warn!("impossible to get poi_subclass for {} because {}", id, e);
+        e.to_string()
+    })?;
+
+    let poi_class = row.try_get("class").map_err(|e| {
+        warn!("impossible to get poi_class for {} because {}", id, e);
+        e.to_string()
+    })?;
+
+    properties.push(Property {
+        key: "poi_subclass".to_string(),
+        value: poi_subclass,
+    });
+
+    properties.push(Property {
+        key: "poi_class".to_string(),
+        value: poi_class,
+    });
+
+    Ok(properties)
+}
+
+fn build_names(langs: &[String], properties: &[Property]) -> Result<mimir::I18nProperties, String> {
+    const NAME_TAG_PREFIX: &str = "name:";
+
+    let properties = properties
+        .iter()
+        .filter_map(|property| {
+            if property.key.starts_with(&NAME_TAG_PREFIX) {
+                let lang = property.key[NAME_TAG_PREFIX.len()..].to_string();
+                if langs.contains(&lang) {
+                    Some(mimir::Property {
```

Comment:
Instead of all the `None`s, you could just return the `Some` here, so we can reduce the code.

Reply:
done